libs/analytics/src/gtm/CowAnalyticsGtm.ts

```diff
@@ -0,0 +1,186 @@
+    }
+  }
+
+  setUserAccount(account: string | undefined): void {
+    this.setContext(AnalyticsContext.userAddress, account || 'disconnected')
+    this.pushToDataLayer({
+      event: 'set_user_account',
+      userId: account || undefined,
+    })
+  }
+
+  sendPageView(path?: string, params?: string[], title?: string): void {
+    this.debouncedPageView(path, params, title)
+  }
+
+  private _sendPageView(path?: string, params?: string[], title?: string): void {
+    this.pushToDataLayer({
+      event: 'page_view',
+      page_path: path,
+      page_title: title,
+      page_params: params,
+      ...this.getDimensions(),
+    })
+  }
+
+  sendEvent(event: string | EventOptions, params?: unknown): void {
+    const eventData: DataLayerEvent =
+      typeof event === 'string'
+        ? { event, ...(params as Record<string, unknown>) }
+        : {
+            // Create specific event name from category and action
+            event: `${event.category.toLowerCase()}_${event.action.toLowerCase().replace(/\s+/g, '_')}`,
+
+            // Core parameters at root level
+            category: event.category,
+            action: event.action,
+            label: event.label,
+            value: event.value,
+
+            // Non-interaction flag
+            non_interaction: event.nonInteraction,
+
+            // Spread dimensions at root level
+            ...this.getDimensions(),
+
+            // Include additional dynamic properties if present
+            ...((event as GtmEvent).orderId && { order_id: (event as GtmEvent).orderId }),
+            ...((event as GtmEvent).orderType && { order_type: (event as GtmEvent).orderType }),
+            ...((event as GtmEvent).tokenSymbol && { token_symbol: (event as GtmEvent).tokenSymbol }),
+            ...((event as GtmEvent).chainId && { chain_id: (event as GtmEvent).chainId }),
+          }
+
+    this.pushToDataLayer(eventData)
+  }
+
+  sendTiming(timingCategory: string, timingVar: string, timingValue: number, timingLabel: string): void {
+    this.pushToDataLayer({
+      event: 'timing_complete',
+      timing_category: timingCategory,
+      timing_variable: timingVar,
+      timing_value: timingValue,
+      timing_label: timingLabel,
+      ...this.getDimensions(),
+    })
+  }
+
+  sendError(error: Error, errorInfo?: string): void {
+    this.pushToDataLayer({
+      event: 'exception',
+      error_description: error.toString() + (errorInfo ? ': ' + errorInfo : ''),
+      error_fatal: true,
+      ...this.getDimensions(),
+    })
+  }
+
+  outboundLink({ label, hitCallback }: OutboundLinkParams): void {
+    this.pushToDataLayer({
+      event: 'outbound_link',
+      outbound_label: label,
+      ...this.getDimensions(),
+    })
+
+    // Execute callback after pushing to dataLayer
+    if (hitCallback) {
+      setTimeout(hitCallback, 0)
+    }
+  }
+
+  setContext(key: AnalyticsContext, value?: string): void {
+    if (value) {
```

Comment:
This is a dangerous condition. The value should be checked as `typeof value !== 'undefined''`. Otherwise, you will delete a dimension if the value is 0 for example.

Reply:
Thanks for catching this. I've implemented your suggested fix using typeof value !== 'undefined' and made the behavior consistent throughout the analytics flow. Falsy values will now be properly preserved. Let me know if you spot anything else.